controllers/onload_controller.go

```diff
@@ -194,41 +199,79 @@ func (r *OnloadReconciler) deleteLabels(ctx context.Context, namespacedName type
 		return nil
 	}
 
-	err := deleteLabel(kmmLabelName(namespacedName.Name, namespacedName.Namespace))
-	if err != nil {
-		return err
+	labels := []string{
+		kmmOnloadLabelName(namespacedName.Name, namespacedName.Namespace),
+		kmmSFCLabelName(namespacedName.Name, namespacedName.Namespace),
+		onloadLabelName(namespacedName.Name, namespacedName.Namespace),
 	}
 
-	err = deleteLabel(onloadLabelName(namespacedName.Name, namespacedName.Namespace))
-	if err != nil {
-		return err
+	for _, label := range labels {
+		err := deleteLabel(label)
+		if err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
 func (r *OnloadReconciler) addKmmLabelsToNodes(ctx context.Context, onload *onloadv1alpha1.Onload) (*ctrl.Result, error) {
 	log := log.FromContext(ctx)
-	kmmPodLabelSet := labels.Set{"kmm.node.kubernetes.io/module.name": onload.Name + onloadModuleNameSuffix}
-
-	labels := labels.FormatLabels(onload.Spec.Selector)
-	labelKey := kmmLabelName(onload.Name, onload.Namespace)
 
-	nodes, err := r.listNodesWithLabels(ctx, labels)
+	// Figure out a list of nodes that match Onload's selector.
+	onloadLabels := labels.FormatLabels(onload.Spec.Selector)
+	nodes, err := r.listNodesWithLabels(ctx, onloadLabels)
 	if err != nil {
 		return nil, err
 	}
 
+	// A list of kmm labels with the corresponding pod labels
+	// to find out if the given module kind is still in use.
+	kmmLabels := []struct {
+		isUsed      bool
+		labelKey    string
+		podLabelSet labels.Set
+	}{
+		{
+			true, // Onload module kind is always used for Onload CR
+			kmmOnloadLabelName(onload.Name, onload.Namespace),
+			labels.Set{"kmm.node.kubernetes.io/module.name": onload.Name + onloadModuleNameSuffix},
+		},
+		{
+			onloadUsesSFC(onload),
+			kmmSFCLabelName(onload.Name, onload.Namespace),
+			labels.Set{"kmm.node.kubernetes.io/module.name": onload.Name + sfcModuleNameSuffix},
+		},
+	}
```

Comment:
Rather than including a conditional bool in the struct, have you considered conditionally adding the the sfc label to the slice?
Something like:
```
kmmLabels = []labels{ Onload definition}
if onloadUsesSFC(onload) {
    kmmLabels = append(kmmLabels, {sfc definition})
}
```

Then you could remove the check for isUsed later.

Reply:
Totally. 👍

Reply:
Re-opening, because there's another similar `isUsed` occurrence in `handleModuleUpdate()`.

Reply:
Yes, this `isUsed` would be non-trivial to remove.